python/hidet/graph/frontend/torch/dynamo_backends.py

```diff
@@ -10,28 +10,65 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # pylint: disable=no-name-in-module
-from typing import List, Callable, Sequence, Union
+from typing import List, Sequence, Union
 import logging
 import torch
 import hidet.option
+from hidet import Tensor
+from hidet.ir import dtypes
 from hidet.ir.type import DataType
 from hidet.ir.expr import SymbolVar
+from hidet.runtime import CompiledGraph
 from hidet.graph.flow_graph import FlowGraph
 from hidet.graph.transforms import PassContext, optimize
-from hidet.runtime import CompiledGraph
 from hidet.cuda.graph import CudaGraphCreationError
-from hidet.ir import dtypes
-from .utils import serialize_output, deserialize_output, resolve_save_dir_multigraph
 from .dynamo_config import dynamo_config
-
+from .interpreter import Interpreter
+from .utils import serialize_output, deserialize_output, resolve_save_dir_multigraph
+from .utils import symbol_like_torch
 
 logger = logging.getLogger(__name__)
 
 
-def generate_executor(flow_graph: FlowGraph) -> Callable:
+def get_flow_graph(interpreter, example_inputs) -> FlowGraph:
+    # prepare dummy and symbolic inputs for correctness and flow graph construction
+    inputs: List[Union[Tensor, SymbolVar, int, bool, float]] = []  # for flow graph construction
+    for example_input in example_inputs:
+        if isinstance(example_input, torch.Tensor):
+            symbolic_input = symbol_like_torch(example_input)
+            inputs.append(symbolic_input)
+        elif isinstance(example_input, (int, bool, float)):
+            inputs.append(example_input)
+        elif isinstance(example_input, torch.SymInt):
+            from torch.fx.experimental.symbolic_shapes import SymNode
+
+            node: SymNode = example_input.node
+            try:
+                inputs.append(node.pytype(example_input))
+            except RuntimeError:
```

Comment:
should we be concerned here? 
RuntimeError seems to be quite generic so it might not just be the error that you think you can handle. I suggest we log a warning down below if it does not happen too often. 

If possible, I prefer that you check the pre-condition that would cause the runtime error and instead of try-except, use 
```
if not (the condition that would trigger runtime error): 
  inputs.append(...)
else:
  # is a symbolic scalar input
```

Reply:
I see that this part is ported from the original code so you might not know why. Please ignore it for now

Reply:
Hi @xinli-git, FYI, we (Allan and I) have tried that, but find out it is hard to find the `the condition that would trigger runtime error`.